packages/zoe/src/typeGuards.js

```diff
@@ -231,11 +231,10 @@ export const SourceBundleShape = M.recordOf(
   M.string(),
   M.string({ stringLengthLimit: Infinity }),
 );
-export const ModuleFormatBundleShape = M.splitRecord(
-  harden({}),
-  harden({ moduleFormat: M.any() }),
+export const BundleShape = M.and(
+  M.partial({ moduleFormat: M.string() }),
+  M.recordOf(M.string(), M.string()),
 );
```

Comment:
I think this was more correct before literally taking @kriskowal 's advice.
   * @Chris-Hibbert , We still need the `{ stringLengthLimit: Infinity }` to evade the limit check, right?
   * @kriskowal , `M.partial` is deprecated in favor of `M.splitArray` and `M.splitRecord`.
   * @kriskowal , By using `M.partial` you're saying that `moduleFormat` is optional, as did the old code when the first argument to `splitRecord` was `harden({})`. But your prose says it is mandatory. Assuming it is mandatory, I removed that first argument to `splitRecord` below.

So, does the following look right?

```js
export const ModuleFormatBundleShape = M.splitRecord(
  harden({ moduleFormat: M.any() }),
);

export const BundleShape = M.or(ModuleFormatBundleShape, SourceBundleShape);
```

Reply:
(just edited my suggested code from `M.or` to `M.and`)

Reply:
> We still need the { stringLengthLimit: Infinity } to evade the limit check, right?

I believe so. It's still in `SourceBundleShape`. I don't think it's necessary anywhere else.

```
export const SourceBundleShape = M.recordOf(
  M.string(),
  M.string({ stringLengthLimit: Infinity }),
);
```

Reply:
I pushed this change, and tests are passing now. 

@kriskowal with your agreement that this change is correct, I'll merge this PR.